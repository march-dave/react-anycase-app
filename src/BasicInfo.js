import React, { Component } from "react";
import { connect } from "react-redux";

const mapStateToProps = state => {
  return {
    GetUserName: state.username
  };
};

class BasicInfo extends Component {
  handleChange = () => {};

  render() {
    return (
      <div>
        <input
          id="username"
          type="text"
          name="username"
          value={this.props.GetUserName}
          readOnly
        />
        <input type="text" onChange={this.handleChange} />
      </div>
    );
  }
}

BasicInfo = connect(
  mapStateToProps,
  null
)(BasicInfo);

export default BasicInfo;
