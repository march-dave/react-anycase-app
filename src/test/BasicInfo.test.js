import React from "react";
<<<<<<< HEAD:src/BasicInfo.test.js
import BasicInfo from "./BasicInfo";

// import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";

// configure({ adapter: new Adapter() });
=======
import BasicInfo from "../BasicInfo";
import { shallow } from "enzyme";
>>>>>>> 162079cac5d7098b085421a12b8df00cf5715b18:src/test/BasicInfo.test.js

let component = null;

describe("BasicInfo Component Load", () => {
  test("Render correctly", () => {
    component = shallow(<BasicInfo />);
  });

  test("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  describe("insert new text", () => {
    test("has an input", () => {
      expect(component.find("input").exists());
    });
  });

  let changed = null;
  const handleChange = name => {
    changed = name;
  };

  // test("handleChange", () => {
  //   const wrapper = shallow(<BasicInfo onChange={handleChange} />);
  //   expect(wrapper.);
  // } )
});
