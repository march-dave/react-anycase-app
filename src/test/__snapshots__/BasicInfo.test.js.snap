// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo Component Load matches snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
